src/commonMain/kotlin/expressions/Var.kt

```diff
@@ -0,0 +1,37 @@
+package expressions
+
+import LogicExpression
+import intValue
+
+internal object Var : LogicExpression {
+    override val key: String = "var"
+
+    override operator fun invoke(expression: Any?, data: Any?): Any? {
+        var value: Any? = data
+        val varName = if (expression is List<*>) {
+            expression.firstOrNull()
+        } else {
+            expression
+        }.toString()
+
+        when (value) {
```

Comment:
I thought that this expression can be changed to
```
val value = when (data) {
            is List<*> -> {
                val indexParts = varName.split(".")
                if (indexParts.size == 1) data[indexParts.first().intValue] else getRecursive(indexParts, data)
            }
            is Map<*, *> -> varName.split(".").forEach {
                (data as? Map<*, *>)?.get(it)
            }
            else -> data
```
But it turns out it makes tests fail. 

I tried refactoring it in. smaller steps and this version works:
![Screenshot 2022-01-28 at 11 39 07](https://user-images.githubusercontent.com/82953872/151532737-b0fd7104-bd59-4a48-a69e-fd4aeda47f2d.png)

but changing value to data in line 23 breaks tests
![Screenshot 2022-01-28 at 11 39 27](https://user-images.githubusercontent.com/82953872/151532781-529ecb13-844c-4b3e-8379-33a3c60cb60a.png)

Do you understand (and pls explain) why it happens?

Reply:
I have refactored this code, check it out and discuss if you need